lib/Image/HtmlToImage.php

```diff
@@ -0,0 +1,202 @@
+    {
+        try {
+
+            $extraHeaders = [
+                'X-Foo' => 'Bar' // required, as extraHttpHeaders() requires at least one entry
+            ];
+
+            if (null !== $sessionId && null !== $sessionName) {
+                $extraHeaders['Cookie'] = $sessionName . '=' . $sessionId;
+            }
+
+            /** @var GotenbergAPI|object $request */
+            $request = GotenbergAPI::chromium(Config::getSystemConfiguration('gotenberg')['base_url']);
+            if(method_exists($request, 'screenshot')) {
+                $request->screenshot()
+                    ->png()
+                    ->extraHttpHeaders($extraHeaders)
+                    ->url($url);
+
+                $file = GotenbergAPI::save($request, PIMCORE_SYSTEM_TEMP_DIRECTORY);
+                return rename(PIMCORE_SYSTEM_TEMP_DIRECTORY . '/' . $file, $outputFile);
+            }
+
+        } catch (\Exception $e) {
+            // nothing to do
+        }
+
+        return false;
+    }
+
+    /**
+     * @throws \Exception
+     */
+    public static function convertChromium(string $url, string $outputFile, ?string $sessionName = null, ?string $sessionId = null, string $windowSize = '1280,1024'): bool
+    {
+        $chromiumUri = \Pimcore\Config::getSystemConfiguration('chromium')['uri'];
+        if (!empty($chromiumUri)) {
+            try {
+                $browser = BrowserFactory::connectToBrowser($chromiumUri);
+            } catch (\Exception $e) {
+                Logger::debug((string) $e);
+
+                return false;
+            }
+        } else {
+            $binary = self::getChromiumBinary();
+            if (!$binary) {
+                return false;
+            }
+            $browserFactory = new BrowserFactory($binary);
+            $browser = $browserFactory->createBrowser([
+                'noSandbox' => file_exists('/.dockerenv'),
+                'startupTimeout' => 120,
+                'windowSize' => explode(',', $windowSize),
+            ]);
+        }
+
+        try {
+            $headers = [];
+            if (null !== $sessionId && null !== $sessionName) {
+                $headers['Cookie'] = $sessionName . '=' . $sessionId;
+            }
+
+            $page = $browser->createPage();
+
+            if (!empty($headers)) {
+                $page->getSession()->sendMessageSync(new Message(
+                    'Network.setExtraHTTPHeaders',
+                    ['headers' => $headers]
+                ));
+            }
+
+            $page->navigate($url)->waitForNavigation();
+
+            $page->screenshot([
+                'captureBeyondViewport' => true,
+                'clip' => $page->getFullPageClip(),
+            ])->saveToFile($outputFile);
+        } catch (\Throwable $e) {
+            Logger::debug('Could not create image from url ' . $url . ': ' . $e);
+
+            return false;
+        } finally {
+            $browser->close();
```

Comment:
```suggestion
            $page->close();
```
should be the same as https://github.com/pimcore/web-to-print-bundle/issues/58